backend/src/components/contact.js

```diff
@@ -59,8 +66,60 @@ async function getRawContactFacilities(contactId) {
   }
 }
 
+async function createContact(req, res) {
+  try {
+    const ccofRoles = (await getRoles()).map((role) => role.data);
+
+    if (req.body.bceid) {
+      const existingContact = (await getOperation(`contacts?$filter=ccof_username eq '${req.body.bceid}' and statecode eq 0`))?.value?.[0];
+
+      if (existingContact) {
+        return res.status(HttpStatus.PRECONDITION_FAILED).json({
+          message: 'A contact with this BCeID already exists.',
+          contactId: existingContact.contactid,
+        });
+      }
+    }
+    const contactPayload = new MappableObjectForBack(req.body, ContactMappings).toJSON();
+    if (req.body.organizationId) {
+      contactPayload['parentcustomerid_account@odata.bind'] = `/accounts(${req.body.organizationId})`;
+    }
+    if (req.body.portalRole) {
```

Comment:
There's inconsistency in the payload format between what we send to the frontend and what's received here in the backend. Ideally they should have have the role attribute as described above. In that case we wouldn't need to lookup the id based on the number as we'd already have it.